### ISSUE TITLE
User "sub" attribute in JWT payload changes after upgrade from 2023.10 to 2024.2

### ISSUE THREAD

**Issue by iRaven4522** [goauthentik/authentik#9361]
**Describe the bug**
When upgrading from version 2023.10.4 to 2024.2.2 (latest at the time of writing), many OIDC/SAML applications including WikiJS, Zabbix, Tailscale, and more detect existing accounts on both ends of authentication as new. The services act like a brand new user has signed in. In many cases either my access permissions don't sync up, or I'm refused authentication on services that don't support the same email on one user.

**To Reproduce**

1. Use authentik 2023.10.X (any version under 2023.10 applies.)
2. Sync some users and groups using an Active Directory connection.
3. Note the "sub" attribute of a user connected with AD before upgrading.
4. Follow general upgrade procedure (in release notes) to upgrade to 2024.2.2.
5. Check the "sub" attribute of a user connected with AD. This should not have changed.

**Expected behavior**
The upgrade and database migration should work as expected with no user identifiers (or the "sub" attribute in particular) changing (that in turn breaks auth on many providers on the services side).

**Screenshots**
Using 2023.10.7:
![Screenshot_20240331_182634](https://github.com/goauthentik/authentik/assets/44326347/8bbd42e6-7c77-4c0d-8c3a-42e9a9e6be7b)
After upgrade (and automatic database migration tasks) to 2023.2.3:
![firefox_3miqKjnZPe](https://github.com/goauthentik/authentik/assets/44326347/b243a446-aa03-45f7-959d-865b7fb53ed0)

**Logs**
N/A

**Version and Deployment (please complete the following information):**

-   authentik version: 2024.2.2
-   Deployment: docker-compose

**Additional context**
I've tried this on a few test instances with the same result, using the database from my production 2023.10 instance.

**Comment by BeryJu** [comment]
This could be due to https://github.com/goauthentik/authentik/pull/8776, as there was a bug that would sometimes generate two install IDs, in which case authentik would pick one non-deterministically. With the PR above authentik will always pick the same one, however in very rare case that might be a different one as before.

Before upgrading you can check the install ID detected by authentik by navigating to `/api/v3//enterprise/license/get_install_id/`, noting which value is returned, and then in the `authentik_install_id` table in the `public` scheme delete the value that isn't returned from that endpoint

**Comment by iRaven4522** [comment]
Yep, that worked. I went through the authentik user with psql and noticed there was an additional install_id value in the ID row in the table. Deleted it, upgraded from 2023.10 to 2024.2 and the sub value is unchanged now (I spun up authentik on a different server VM to not affect my production setup). 
I'm currently upgrading to the latest at the time of writing (2024.4.2) to see if it happens again with that migration.
